src/invoices/InvoiceDetails/hooks/useHasPendingOrders.js

```diff
@@ -0,0 +1,33 @@
+import { useMemo } from 'react';
+import { ORDER_STATUSES } from '@folio/stripes-acq-components';
+import { useOrders } from '../../../common/hooks';
+
+export const useHasPendingOrders = (orderLinesMap = {}) => {
+  const orderIds = useMemo(() => {
+    return Object.values(orderLinesMap)
+      .map(({ purchaseOrderId }) => purchaseOrderId);
+  }, [orderLinesMap]);
+
+  const {
+    isLoading,
+    orders,
+    isFetched,
+  } = useOrders(orderIds);
+
+  const hasPendingOrders = useMemo(() => {
+    if (isLoading) {
```

Comment:
it's strange  login to say that there are pending orders when loading in progress, why not to disable action when it's loading?

Reply:
What do you mean by "disable action"? Do you mean the Pay/Approve button should be disabled?

Reply:
actually this condition `return true` will disable the action as the name suggests `hasPendingOrders`.

Reply:
why do you need to have this strange logic when you can put isLoading to condition to disabled action?

Reply:
your method should look like
```
const hasPendingOrders = useMemo(() => {
  return orders.some(({ workflowStatus }) => workflowStatus === ORDER_STATUSES.pending);
});

Reply:
@NikitaSedyx What do you think about this?
```
const hasPendingOrders = useMemo(() => {
    return orders.some(({ workflowStatus }) => workflowStatus === ORDER_STATUSES.pending);
  }, [orders]);

  return {
    isLoading,
    hasPendingOrders: isLoading || hasPendingOrders,
  };
```
so that the logic will stay the same `isApprovePayEnabled={!isVendorInactive && !hasPendingOrders}` https://github.com/folio-org/ui-invoice/pull/694#discussion_r1212850557

Reply:
isLoading is not hasPendingOrders, please do not mix, it's not transparent

Reply:
isLoading is not hasPendingOrders, please do not mix, it's not transparent